doc/developers/developing_estimators.rst

```diff
@@ -0,0 +1,761 @@
+.. _developing_estimators:
```

Comment:
This section is intended to be about development in general, not just developing estimator. Maybe change the ref/file/section to e.g. "Developing scikit-learn" ? (not great either, others will have better ideas)